apps/noaatides/noaa_tides.star

```diff
@@ -0,0 +1,304 @@
+"""
+Applet: NOAA Tides
+Summary: Display NOAA Tides
+Description: Display daily tides from NOAA stations.
+Author: tavdog
+"""
+
+production = True
+debug = False
+print_debug = False
+
+load("render.star", "render")
+load("schema.star", "schema")
+load("http.star", "http")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+load("xpath.star", "xpath")
+load("re.star", "re")
+
+default_location = """
+  {
+	"lat": "20.8911",
+	"lng": "-156.5047",
+	"description": "Wailuku, HI, USA",
+	"locality": "Maui",
+	"timezone": "America/Honolulu"
+}
+"""
+
+def debug_print(arg):
+    if print_debug:
+        print(arg)
+
+def get_stations(location):  # assume we have a valid location dict
+    stations_json = {}
+    station_options = list()
+    url = "https://api.tidesandcurrents.noaa.gov/mdapi/prod/webapi/tidepredstations.json?lat=%s&lon=%s&radius=50" % (location["lat"], location["lng"])
```

Comment:
Would you mind adding some code to  truncate lat/lng before doing this call? We do this to avoid unnecessarily leaking/sharing the user's exact location. I'm thinking 1 or at most 2 decimal points would be sufficient for this application. You can find some info on how this affects accuracy [over here](https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude), and a few examples of how other apps handle it [over here](https://github.com/tidbyt/community/search?q=truncate).

Reply:
ok, all done.